frontend/src/plugins/impl/MultiselectPlugin.tsx

```diff
@@ -53,6 +54,24 @@ interface MultiselectProps extends Data {
 
 const Multiselect = (props: MultiselectProps): JSX.Element => {
   const id = useId();
+  const [filteredOptions, setFilteredOptions] = useState(props.options);
+
+  useEffect(() => {
+    setFilteredOptions(props.options);
+    return () => setFilteredOptions([]);
+  }, [props.options]);
+
+  const filterOptions = (search: string) => {
+    if (!search) {
+      setFilteredOptions(props.options);
+      return;
+    }
+    setFilteredOptions(
+      props.options.filter(
+        (option) => multiselectFilterFn(option, search) === 1,
+      ),
+    );
+  };
```

Comment:
the react team suggests minimizing the use of useEffect, you could re-write this as:

```ts
  const [searchQuery, setSearchQuery] = useState<string>("");
  const filteredResults = useMemo(() => {
      if (!searchQuery) {
         return props.options;
      }
      return props.options.filter(
          (option) => multiselectFilterFn(option, search) === 1,
      );
  }, [props.options, searchQuery]);
```

Reply:
TIL, thanks for letting me know